### ISSUE TITLE
Issue with Type in Aot Build 

### ISSUE THREAD

**Issue by elvirdolic** [ngrx/platform#63]
Hello,

Versions:
typescript: 2.4.1
ngrx/store: latest nightly build
angular: 4.2.5

On AOT Build im receiving type errors

```
Type 'Store<
Customer[]>' is not assignable to type 'Observable<Customer[]>'.
  Types of property 'subscribe' are incompatible.
    Type '{ (): Subscription; (observer: PartialObserver<Readonly<Customer[]>>): Subscription; (next?: (val...' is not assignable to type '{ (): Subscription; (observer: Pa
rtialObserver<Customer[]>): Subscription; (next?: (value: Custom...'.
      Types of parameters 'next' and 'next' are incompatible.
        Types of parameters 'value' and 'value' are incompatible.
          Type 'Readonly<Customer[]>' is not assignable to type 'Customer[]'.
```

in customers-list.component
```
customers$: Observable<Customer[]>; 
this.customers$ = this.store.select(customers);

```
customers selector

```
export const customers = createSelector(customerState, (state: state) => {
    return state.customers.ids.map(id => state.customers.entities[id]);
});

```

**Comment by agugan** [comment]
#46  Could be due to part explained here.

**Comment by jinder** [comment]
@elvirdolic @agugan Yes, this is a dupe of #46.

**Comment by MikeRyanDev** [comment]
Closes in #72
